rdsbroker/broker.go

```diff
@@ -387,20 +391,79 @@ func (b *RDSBroker) LastOperation(instanceID string) (brokerapi.LastOperationRes
 	return lastOperationResponse, nil
 }
 
+func (b *RDSBroker) CheckAndRotateCredentials() error {
+	dbInstanceDetailsList, err := b.dbInstance.DescribeByTag("Broker Name", b.brokerName)
+	if err != nil {
+		b.logger.Error("Could not retrieve the list of DB instances", err)
+		return err
+	}
+
+	for _, dbInstanceDetails := range dbInstanceDetailsList {
+		dbAddress := dbInstanceDetails.Address
+		dbPort := dbInstanceDetails.Port
+		masterUsername := dbInstanceDetails.MasterUsername
+		serviceInstanceID := b.dbInstanceIdentifierToServiceInstanceID(dbInstanceDetails.Identifier)
+
+		var dbName string
+		if dbInstanceDetails.DBName != "" {
+			dbName = dbInstanceDetails.DBName
+		} else {
+			dbName = b.dbName(dbInstanceDetails.Identifier)
+		}
+
+		sqlEngine, err := b.sqlProvider.GetSQLEngine(dbInstanceDetails.Engine)
+		if err != nil {
+			b.logger.Error("Could not get the SQL Engine", err)
+			return err
+		}
+
+		err = sqlEngine.Open(dbAddress, dbPort, dbName, masterUsername, b.masterPassword(serviceInstanceID))
+		if err != nil {
+			b.logger.Error("Error opening connection to the database", err)
+			return err
+		} else {
+			defer sqlEngine.Close()
+		}
+
+		// Open may not validate the connection
+		err = sqlEngine.Ping()
+		if err != nil {
+			if strings.Contains(err.Error(), "Login failed") {
+				b.logger.Error(fmt.Sprintf("Login failed when connecting to DB %v at %v, will attempt to reset the password", dbName, dbAddress), err)
```

Comment:
I think this should be an "info" log message rather than "error" because it's an expected action if we've rotated the seed.